worlds/AutoSNIClient.py

```diff
@@ -15,6 +20,19 @@ def __new__(cls, name: str, bases: Tuple[type, ...], dct: Dict[str, Any]) -> Aut
         new_class = super().__new__(cls, name, bases, dct)
         if "game" in dct:
             AutoSNIClientRegister.game_handlers[dct["game"]] = new_class()
+
+        if "patch_suffix" in dct:
+            if dct["patch_suffix"] is not None:
+                existing_identifier: SuffixIdentifier = component.file_identifier
+                new_suffixes = [*existing_identifier.suffixes]
+
+                if type(dct["patch_suffix"]) is str:
```

Comment:
```suggestion
                if isinstance(dct["patch_suffix"], str):
```
If someone uses a subclass of `str`, the `type()` type check wouldn't catch it, and it would cause a bug that could be difficult to figure out (because it wouldn't crash), when there's no reason for it to be a bug at all.

`isinstance` is usually the preferred way to do a type check.